lib/ruby_lsp/test_reporter.rb

```diff
@@ -67,5 +76,28 @@ def send_message(method_name, params)
         $stdout.write("Content-Length: #{json_message.bytesize}\r\n\r\n#{json_message}")
       end
     end
+
+    class IOWrapper < SimpleDelegator
+      #: (Array[String]) -> void
+      def puts(*args)
+        args.each { |arg| log("#{arg}\n") }
+      end
+
+      #: (Array[String]) -> void
+      def print(*args)
+        args.each { |arg| log(arg.to_s) }
+      end
+
+      private
+
+      #: (String) -> void
+      def log(message)
+        TestReporter.append_output(message: message)
+      end
+    end
   end
 end
+
+# We wrap the default output stream so that we can capture anything written to stdout and emit it as part of
+# the JSON event stream.
+$> = RubyLsp::TestReporter::IOWrapper.new($stdout)
```

Comment:
We still need to keep a handle for the original stdout and replace it with the same wrapper.